app/cdap/components/shared/ConfigurationGroup/index.tsx

```diff
@@ -140,7 +146,27 @@ const ConfigurationGroupView: React.FC<IConfigurationGroupProps> = ({
     setFilteredConfigurationGroups(newFilteredConfigurationGroup);
     getOrphanedErrors();
   }
-
+  // Some fields may have been cleared and lost their default value
+  // this function adds default value to property which shows and
+  // currently has a null value
+  function addDefaultValueToShownProperties(configGroup, newValues) {
+    let isUpdated = false;
+    configGroup.map((config) => {
+      config.properties.map((property) => {
+        if (property.show && newValues[property.name] === null) {
+          newValues[property.name] = property.defaultValue;
+          isUpdated = true;
+        }
+      });
+    });
+    // some field have chaining conditions, so updated values may
+    // change some conditions result, hence continuously update
+    // (Is it going to create infinite loop?)
+    if (isUpdated) {
+      updateFilteredConfigurationGroup(configGroup, newValues);
```

Comment:
you're calling the same function this function is called from if its updated? this logic is super weird to me, couldn't this create an infinite loop?

Reply:
I thought about that. But the case seems like I need to continuously update it till I cannot do it anymore. I can't think of anything other than recursion.

Reply:
Yea I encountered infinite loop cases... I need to think about it more